### ISSUE TITLE
Unhandled exception when choosing TheBoy181 Repo

### ISSUE THREAD

**Issue by mp44christos** [amakvana/YuzuModDownloader#21]
When selecting the download server “TheBoy181 GitHub Repo” a Microsoft .NET Framework error pops up and the app crashes.
The app works perfectly with the official Repo.

**Screenshots**
https://imgur.com/a/aRu58SO
https://pastebin.com/vf6aQ9mu

**Desktop**
 - OS: Windows 11 Pro Insider Preview
Version	22H2
Installed on	‎20/‎12/‎2022
OS build	25267.1000
Experience	Windows Feature Experience Pack 1000.25267.1000.0
 - Yuzu Version mainline 1277
 - Yuzu Installed
 - Yuzu Mod Downloader Version 1.3.0.0
 - Installed Microsoft Visual C++ from the file provided by the guide
 - Currently installed "classic" .NET Versions in the system:
2.0.50727.4927 Service Pack 2
3.0.30729.4926 Service Pack 2
3.5.30729.4926 Service Pack 1
4.0.0.0
4.8.09032

**Comment by amakvana** [comment]
Can't seem to reproduce it on my end.

Can you send me the list of games you have within Yuzu, along with their Game ID's? The ID's begin with 0x, see screenshot below:

![image](https://user-images.githubusercontent.com/66264136/208900123-36f41acb-daf5-4c7a-af08-3dcbc25e5f73.png)

Also, can you send me a screenshot of your "Filesystem" setup? Open Yuzu > Emulation > Configure > System > Filesystem. It should look like this:

![image](https://user-images.githubusercontent.com/66264136/208900478-e3e71d42-e852-4fdf-adce-791f857ae473.png)

**Comment by mp44christos** [comment]
0100A3D008C5C000
010028600EBDA000
0100000000010000
01007EF00011E000
01006F8002326000
01007300020FA000
01004A4010FEA000
01009BF0072D4000
0100F9F00C696000
0100C1F0051B6000
01002B00111A2000
01004D300C5AE000
0100DCA0064A6000
010067300059A000
0100317013770000
0100152000022000
01006FE013472000
010093801237C000
0100F4C009322000
01001F5010DFA000
0100187003A36000
0100C2500FC20000
01006A800016E000
01006BB00C6F0000
01006000040C2000

https://imgur.com/a/JV2jk7M

**Comment by reloxx13** [comment]
+1    

```
Informationen über das Aufrufen von JIT-Debuggen
anstelle dieses Dialogfelds finden Sie am Ende dieser Meldung.

************** Ausnahmetext **************
System.Net.WebException: Die Anfrage wurde abgebrochen: Die Verbindung wurde unerwartet getrennt..
   bei System.Net.ConnectStream.Read(Byte[] buffer, Int32 offset, Int32 size)
   bei System.Net.HttpWebRequest.MakeMemoryStream(Stream stream)
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei YuzuModDownloader.ModDownloader.<DownloadModsAsync>d__26.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei YuzuModDownloader.TheBoy181ModDownloader.<DownloadModsAsync>d__4.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   bei YuzuModDownloader.frmMain.<btnDownload_Click>d__2.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)


************** Geladene Assemblys **************
mscorlib
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4515.0 built by: NET48REL1LAST_C.
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll.
----------------------------------------
YuzuModDownloader
    Assembly-Version: 1.3.0.0.
    Win32-Version: 1.3.0.0.
    CodeBase: file:///A:/Emus/yuzu/app/Yuzu%20Early%20Access/YuzuModDownloader.exe.
----------------------------------------
System.Windows.Forms
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4550.0 built by: NET48REL1LAST_C.
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll.
----------------------------------------
System
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4536.0 built by: NET48REL1LAST_C.
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll.
----------------------------------------
System.Drawing
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4390.0 built by: NET48REL1LAST_C.
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll.
----------------------------------------
System.Configuration
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4190.0 built by: NET48REL1LAST_B.
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll.
----------------------------------------
System.Core
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4590.0 built by: NET48REL1LAST_B.
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll.
----------------------------------------
System.Xml
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0 built by: NET48REL1.
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll.
----------------------------------------
Accessibility
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0 built by: NET48REL1.
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/Accessibility/v4.0_4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll.
----------------------------------------
HtmlAgilityPack
    Assembly-Version: 1.11.46.0.
    Win32-Version: 1.11.46.
    CodeBase: file:///A:/Emus/yuzu/app/Yuzu%20Early%20Access/HtmlAgilityPack.DLL.
----------------------------------------
System.Net.Http
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0 built by: NET48REL1.
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Net.Http/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Net.Http.dll.
----------------------------------------
System.resources
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0 built by: NET48REL1.
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.resources/v4.0_4.0.0.0_de_b77a5c561934e089/System.resources.dll.
----------------------------------------
mscorlib.resources
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0 built by: NET48REL1.
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/mscorlib.resources/v4.0_4.0.0.0_de_b77a5c561934e089/mscorlib.resources.dll.
----------------------------------------
System.Windows.Forms.resources
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0 built by: NET48REL1.
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms.resources/v4.0_4.0.0.0_de_b77a5c561934e089/System.Windows.Forms.resources.dll.
----------------------------------------

************** JIT-Debuggen **************
Um das JIT-Debuggen (Just-In-Time) zu aktivieren, muss in der
Konfigurationsdatei der Anwendung oder des Computers
(machine.config) der jitDebugging-Wert im Abschnitt system.windows.forms festgelegt werden.
Die Anwendung muss mit aktiviertem Debuggen kompiliert werden.

Zum Beispiel:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

Wenn das JIT-Debuggen aktiviert ist, werden alle nicht behandelten
Ausnahmen an den JIT-Debugger gesendet, der auf dem
Computer registriert ist, und nicht in diesem Dialogfeld behandelt.

```

**Comment by amakvana** [comment]
Thanks for this, looks like GitHub handles concurrent connections a little differently. Will look into rolling a fix out as soon as possible

**Comment by amakvana** [comment]
Looking the XML's and filepaths, it's definitely to do with certain characters within the URLs. Going to look into a couple fixes now

**Comment by amakvana** [comment]
@reloxx13 @mp44christos try the copy attached and let me know if this resolves the issue
[YMD_Crash_Fix.zip](https://github.com/amakvana/YuzuModDownloader/files/10278771/YMD_Crash_Fix.zip)

Extract and override the existing YuzuModDownloader copy you both have

**Comment by mp44christos** [comment]
> @reloxx13 @mp44christos try the copy attached and let me know if this resolves the issue [YMD_Crash_Fix.zip](https://github.com/amakvana/YuzuModDownloader/files/10278771/YMD_Crash_Fix.zip)
> 
> Extract and override the existing YuzuModDownloader copy you both have

It worked. But it only downloaded mods for 3 games.

**Comment by amakvana** [comment]
> It worked. But it only downloaded mods for 3 games.

You may have to check what version your games are on. Not all versions of games are compatible with the mods on `theboy181's` server.

I tested your games library in the [reply above](https://github.com/amakvana/YuzuModDownloader/issues/21#issuecomment-1361284255) with all game versions set as 1.0.0 and it downloaded for 5 games.

**Comment by reloxx13** [comment]
heya, thank you for the fast fix, it works now and downloaded alot mods for 6 games from theboy which is a valid count.

**Comment by amakvana** [comment]
Thanks for confirming @reloxx13 @mp44christos. I will push this shortly to the main repo

**Comment by amakvana** [comment]
Closed - Fixed in v1.3.1.0
https://github.com/amakvana/YuzuModDownloader/releases/latest
